### DISCUSSION TITLE
MightyCore, MegaCore and ArduinoOTA

### DISCUSSION THREAD

**Discussion by MCUdude** [JAndrassy/ArduinoOTA#79] [Show and tell]
Hi @jandrassy

I'll just want to let you know that I'm working on rebuilding all the 64kiB+ bootloaders to support flash page copy. This means that all MightyCore and MegaCore compatible chips with 64kiB or more will work with this library out of the box 🙂

**Comment by JAndrassy** [comment]
Thank you Hans.  It will make it much simpler for the AVR users of ArduinoOTA library.
I change the README accordingly after new release of the cores.

Juraj A.

**Reply by MCUdude** [reply]
Excellent! I'm not quite ready to release a new version yet. I'm still working on the new [Flash library](https://github.com/MCUdude/MightyCore/tree/master/avr/libraries/Optiboot_flasher) that would make the Optiboot read/write functionality much more accessible to the average user. You should give it a try! I'm very open to feedback and ideas on what other examples I should provide to demonstrate its functionality.

**Reply by JAndrassy** [reply]
yes I follow the "Questions about Optiboot Flasher" 'issue'. On MCUs without EEPROM the eeprom emulation uses flash, but the do_spm in Optoboot targets MCUs which all have EEPROM a.f.a.i.k. So using flash as storage is useful only for data larger than can be stored in EEPROM. And then there is flash wearing so the users must be aware of this limit to not to write to it too frequently. so I didn't come up with an appropriate example

**Comment by MCUdude** [comment]
A new MegaCore release, v2.1.2 is now available! A new MightyCore release will probably be reasy tomorrow when the CI have finished the build.

**Comment by JAndrassy** [comment]
I updated the README.

For ArduinoOTA these lines for platform.txt are used:
```
## Create output (bin file)
recipe.objcopy.bin.pattern="{compiler.path}{compiler.elf2hex.cmd}" -O binary {compiler.elf2hex.extra_flags} "{build.path}/{build.project_name}.elf" "{build.path}/{build.project_name}.bin"

tools.avrdude.network_cmd={runtime.tools.arduinoOTA.path}/bin/arduinoOTA
tools.avrdude.upload.network_pattern="{network_cmd}" -address {serial.port} -port 65280 -username arduino -password "{network.password}" -sketch "{build.path}/{build.project_name}.bin" -upload /sketch -b
```
the `tools.avrdude.upload` is used only if network upload is executed, but the `recipe.objcopy.bin.pattern` command builds a .bin alongside the .hex on very build. ArduinoOTA takes .bin not only for upload in the `tools.avrdude.upload` but the user can put the binary on a server and then download it from sketch as shown in [Advanced examples](https://github.com/jandrassy/ArduinoOTA/tree/master/examples/Advanced).

README instructs user to copy extras/avr/platform.local.txt next to platform.txt in core to have these commands, but it would be simpler for the users if the core already had the commands in platform.txt
